misc/images/materialized/entrypoint.sh

```diff
@@ -93,8 +93,14 @@ if [ -n "${MZ_RESTART_ON_FAILURE:-}" ]; then
         else
             code=$?
         fi
-        echo "environmentd exited (code: $code); restarting in 5s..." >&2
-        sleep 5
+
+        if [[ $code != 0 ]]; then
+            echo "environmentd exited (code: $code); restarting in 5s..." >&2
+            sleep 5
+        else
+            echo "environmentd exited gracefully; not restarting" >&2
+            exit 0
```

Comment:
I like this because it means `MZ_RESTART_ON_FAILURE` actually functions like restart only on failure, like the name implies.